ds3_java_cli/src/main/java/com/spectralogic/ds3cli/ArgumentFactory.java

```diff
@@ -0,0 +1,128 @@
+import java.util.List;
+import java.util.Properties;
+
+public class ArgumentFactory {
+
+    // always added and read in first parsing
+    public final static Option endpoint = new Option("e", true, "The ds3 endpoint to connect to or have \"DS3_ENDPOINT\" set as an environment variable.");
+    public final static Option accessKey = new Option("a", true, "Access Key ID or have \"DS3_ACCESS_KEY\" set as an environment variable");
+    public final static Option secretKey = new Option("k", true, "Secret access key or have \"DS3_SECRET_KEY\" set as an environment variable");
+    public final static Option command = new Option("c", true, "The Command to execute.  For Possible values, use '--help list_commands.'" );
+    public final static Option proxy = new Option("x", true, "The URL of the proxy server to use or have \"http_proxy\" set as an environment variable");
+    public final static Option http = Option.builder().longOpt("http").desc("Send all requests over standard http").build();
+    public final static Option insecure = Option.builder().longOpt("insecure").desc("Ignore ssl certificate verification").build();
+    public final static Option printHelp = new Option("h", "Help Menu");
+    public final static Option commandHelp = Option.builder()
+            .longOpt("help")
+            .desc("Command Help (provide command name from -c)")
+            .optionalArg(true)
+            .numberOfArgs(1)
+            .build();
+    public static final Option verbose = Option.builder().longOpt("verbose").desc("Log output to console.").build();
+    public static final Option debug = Option.builder().longOpt("debug").desc("Debug (more verbose) output to console.").build();
+    public static final Option trace = Option.builder().longOpt("trace").desc("Trace (most verbose) output to console.").build();
+    public static final Option logVerbose = Option.builder().longOpt("log-verbose").desc("Log output to log file.").build();
+    public static final Option logDebug = Option.builder().longOpt("log-debug").desc("Debug (more verbose) output to log file.").build();
+    public static final Option logTrace = Option.builder().longOpt("log-trace").desc("Trace (most verbose) output to log file.").build();
+    public static final Option printVersion = Option.builder().longOpt("version").desc("Print version information").build();
+    public static final Option viewType = Option.builder().longOpt("output-format")
+            .desc("Configure how the output should be displayed.  Possible values: [" + ViewType.valuesString() + "]").build();
+    public static final Option retries = new Option("r", true,
+            "Specifies how many times puts and gets will be attempted before failing the request. The default is 5");
+    public static final Option bufferSize = new Option("bs", true, "Set the buffer size in bytes. The defalut is 1MB");
+
+    // added as needed by commands
+    public static final Option bucket = Option.builder("b").hasArg(true)
```

Comment:
We should probably call these pre-defined ones, and possibly split it out into a different class to help keep the values segregated and clearer that they mean different things.